src/headless/converse-muc.js

```diff
@@ -25,14 +25,17 @@ converse.MUC_ROLE_CHANGES = ['op', 'deop', 'voice', 'mute'];
 
 const ACTION_INFO_CODES = ['301', '303', '333', '307', '321', '322'];
 
-const MUC_INFO_CODES = {
+converse.MUC_INFO_CODES = {
     'visibility_changes': [100, 102, 103, 172, 173, 174],
     'self': [110],
     'non_privacy_changes': [104, 201],
     'muc_logging_changes': [170, 171],
     'nickname_changes': [210, 303],
     'disconnect_messages': [301, 307, 321, 322, 332, 333],
-}
+    'affiliation_changes': 'affiliation_changes',
+    'join_leave_events': 'join_leave_events',
+    'role_changes': 'role_changes',
```

Comment:
I was more thinking along the lines of:

`'role_changes': [...converse.MUC_ROLE_CHANGES]`

instead of:

`'role_changes': 'role_changes'`

What do you think of that?

Reply:
@jcbrand I went ahead with your suggestion and added values for the three categories. Now visibility can be controlled case by case :)